api/processor_api.py

```diff
@@ -0,0 +1,97 @@
+    :param engine_instance:  A ConfigEngine object which store all of the config parameters. Access to any parameter
+        is possible by calling get_section_dict method.
+    """
+
+    def __init__(self, config):
+        self.config = config
+        self._setup_queues()
+        self._host = self.config.get_section_dict("API")["Host"]
+        self._port = int(self.config.get_section_dict("API")["Port"])
+        self.app = self.create_fastapi_app()
+
+    def _setup_queues(self):
+        QueueManager.register('get_cmd_queue')
+        QueueManager.register('get_result_queue')
+        self._queue_host = self.config.get_section_dict("CORE")["Host"]
+        self._queue_port = int(self.config.get_section_dict("CORE")["QueuePort"])
+        auth_key = self.config.get_section_dict("CORE")["QueueAuthKey"]
+        self._queue_manager = QueueManager(address=(self._queue_host, self._queue_port), authkey=auth_key.encode('ascii'))
+        
+        while True:
+            try:
+                self._queue_manager.connect()
+                break
+            except ConnectionRefusedError:
+                logger.warning("Waiting for core's queue to initiate ... ")
+                time.sleep(1)
+
+        logger.info("Connection established to Core's queue")
+        self._cmd_queue = self._queue_manager.get_cmd_queue()
+        self._result_queue = self._queue_manager.get_result_queue()
+
+    def create_fastapi_app(self):
+        # Create and return a fastapi instance
+        app = FastAPI()
+
+        if os.environ.get('DEV_ALLOW_ALL_ORIGINS', False):
+            # This option allows React development server (which is served on another port, like 3000) to proxy requests
+            # to this server.
+            # WARNING: read this before enabling it in your environment:
+            # https://medium.com/@stestagg/stealing-secrets-from-developers-using-websockets-254f98d577a0
+            from fastapi.middleware.cors import CORSMiddleware
+            app.add_middleware(CORSMiddleware, allow_origins='*', allow_credentials=True, allow_methods=['*'],
+                               allow_headers=['*'])
+        app.mount("/static", StaticFiles(directory="/repo/data/web_gui/static"), name="static")
+
+        @app.get("/restart-engine")
+        async def restart_engine():
+            logger.info("restart-engine requests on api")
+            self._cmd_queue.put(Commands.RESTART)
+            logger.info("waiting for core's response...")
+            result = self._result_queue.get()
+            return result
+        
+        @app.get("/process-video-cfg")
+        async def restart_engine():
+            logger.info("process-video-cfg requests on api")
+            self._cmd_queue.put(Commands.PROCESS_VIDEO_CFG)
+            logger.info("waiting for core's response...")
+            result = self._result_queue.get()
+            return result
+        
+        @app.get("/stop-process-video")
```

Comment:
I have implemented get-config and set-config methods in my [new-arch branch](https://github.com/emma-w-dev/smart-social-distancing/tree/new-arch),  I suggest to merge our codes first and then send a PR on the main repo. @JsonSadler

Reply:
Great!! I just made it [https://github.com/emma-w-dev/smart-social-distancing/pull/2/](https://github.com/emma-w-dev/smart-social-distancing/pull/2/) and resolved the conflicts, assuming you were using your amd64-usbtpu config file, hope I have not messed up anything in there. Please review and make the PR.